### PR TITLE
Add -patches branch when it does not exist

### PR DESCRIPTION
**Pull request by javierpena**
When the -patches branch is not present in the repository, we may want
to create it automatically. If that is the case, we will create the
branch based on the commit received, then push it.

This commit introduces a new setting in patch_rebaser.ini, to enable
the feature if needed. It also honors the dev_mode setting, to avoid
pushing during development.

### CODE REVIEW

File: patch_rebaser/patch_rebaser.py [hunk 1]
```diff
@@ -35,6 +35,40 @@ def find_patches_branch(repo, remote, distgit_branch):
     return None
 
 
+def create_patches_branch(repo, commit, remote, dev_mode=True):
+    """Create new patches branch from commit"""
+    branch_name = os.environ.get('PATCHES_BRANCH', None)
```
Comment (javierpena) [inline patch_rebaser/patch_rebaser.py]: Here I'm taking advantage of the PATCHES_BRANCH env var created by the patch_rebaser.sh script. I think a better idea would be to add a new parameter to the ini file with the patches branch.

That would create some side-effects, like not having to guess the -patches branch in find_patches_branch(), but that should be handled by a separate commit.

WDYT?

Comment (javierpena) [reply patch_rebaser/patch_rebaser.py]: Ack. We are assuming many other env vars are present (all the DLRN_* ones), so let's use it.

File: tests/test_patch_rebaser.py [hunk 2]
```diff
@@ -410,3 +413,87 @@ def test_parse_gerrit_remote_url_https_noport():
...
     assert project == 'base/name'
+
+
+def test_rebaser_missing_patches_branch_no_create(mock_env, mock_config,
+                                                  monkeypatch):
+    """
+    GIVEN a valid patch_rebaser configuration and environment
+    WHEN main() is called
+    WHEN the -patches branch is missing
+    WHEN create_patches_branch is set to false
```
[the remark this answers is not in the document]
Comment (javierpena) [reply tests/test_patch_rebaser.py]: Oh, I misread the other unit tests. It's an easy fix, give me a minute.

[the remark this answers is not in the document]
Comment (jpichon) [reply tests/test_patch_rebaser.py]: LGTM! Thanks
